devito/passes/iet/misc.py

```diff
@@ -90,23 +93,96 @@ def relax_incr_dimensions(iet, **kwargs):
         outer, inner = split(iterations, lambda i: not i.dim.parent.is_Block)
 
         # Get root's `symbolic_max` out of each outer Dimension
+        roots_min = {i.dim.root: i.symbolic_min for i in outer}
         roots_max = {i.dim.root: i.symbolic_max for i in outer}
 
         # Process inner iterations and adjust their bounds
-        for n, i in enumerate(inner):
-            # The Iteration's maximum is the MIN of (a) the `symbolic_max` of current
-            # Iteration e.g. `x0_blk0 + x0_blk0_size - 1` and (b) the `symbolic_max`
-            # of the current Iteration's root Dimension e.g. `x_M`. The generated
-            # maximum will be `MIN(x0_blk0 + x0_blk0_size - 1, x_M)
+        for i in inner:
+            # Usually the Iteration's maximum is the MIN of (a) the `symbolic_max` of
+            # current Iteration e.g. `x0_blk0 + x0_blk0_size - 1` and (b) the
+            # `symbolic_max` of the current Iteration's root Dimension e.g. `x_M`. The
+            # generated maximum will be `MIN(x0_blk0 + x0_blk0_size - 1, x_M)
 
-            # In some corner cases an offset may be added (e.g. after CIRE passes)
+            # There are cases were optimizations may add an offset (e.g. CIRE passes)
```

Comment:
Why "There are cases"... I mean, it's always the MIN of the same two quantities, but one quantity carries implicitly an offset, no...?

That is, the way it's written here, it's like there are two different ways of computing it